File: tools/scan_bs4.py
import os, re

views = 'app/Views'
patterns = {
    'data-dismiss': re.compile(r'data-dismiss=["\']modal'),
    'form-group': re.compile(r'class="form-group"'),
    'close-btn': re.compile(r'class="close"'),
    'dt-cdn-old': re.compile(r'datatables\.net/1\.(11|10|9|8|7)'),
    'dt-bootstrap4': re.compile(r'dataTables\.bootstrap4|buttons\.bootstrap4'),
}

results = {k: [] for k in patterns}

for root, dirs, files in os.walk(views):
    dirs[:] = [d for d in dirs if d not in ['demo', 'emails', 'Examples']]
    for f in files:
        if not f.endswith('.php'):
            continue
        path = os.path.join(root, f)
        try:
            content = open(path, encoding='utf-8', errors='ignore').read()
        except Exception:
            continue
        for k, pat in patterns.items():
            if pat.search(content):
                results[k].append(path)

for k, files in results.items():
    print(f'=== {k} ({len(files)} files) ===')
    for f in sorted(files):
        print(f'  {f}')
    print()

File: tools/scan_bs4.php
<?php

$views = 'app/Views';
$patterns = [
    'form-group'        => '/class="form-group"/',
    'data-toggle'       => '/data-toggle="/',
    'data-target'       => '/data-target="/',
    'sr-only BS4'       => '/class="sr-only"/',
    'data-placement'    => '/data-placement="/',
    'dt-local-load'     => '/<script[^>]+datatables\.net/',
    'modal-show-bs4'    => '/\$\([\'"][^"\']+[\'"]\)\.modal\([\'"]show[\'"]\)/',
    'modal-hide-bs4'    => '/\$\([\'"][^"\']+[\'"]\)\.modal\([\'"]hide[\'"]\)/',
    'float-dt-CSS'      => '/dataTables_(paginate|filter)[^}]+float\s*:/',
];

$results = array_fill_keys(array_keys($patterns), []);

$dirIter = new RecursiveIteratorIterator(
    new RecursiveDirectoryIterator($views, RecursiveDirectoryIterator::SKIP_DOTS)
);

$skip = ['demo', 'emails', 'Examples'];

foreach ($dirIter as $file) {
    if ($file->getExtension() !== 'php') continue;
    $path = $file->getPathname();
    $skip_file = false;
    foreach ($skip as $s) {
        if (strpos($path, DIRECTORY_SEPARATOR . $s . DIRECTORY_SEPARATOR) !== false) {
            $skip_file = true; break;
        }
    }
    if ($skip_file) continue;

    $content = file_get_contents($path);
    if ($content === false) continue;

    foreach ($patterns as $k => $pat) {
        if (preg_match($pat, $content)) {
            $results[$k][] = str_replace(['app\\Views\\', 'app/Views/'], '', $path);
        }
    }
}

foreach ($results as $k => $files) {
    if (empty($files)) continue;
    echo "=== $k (" . count($files) . " files) ===\n";
    sort($files);
    foreach ($files as $f) {
        echo "  $f\n";
    }
    echo "\n";
}
